app/backend/approaches/approach.py

```diff
@@ -116,6 +117,7 @@ def __init__(
         self.openai_host = openai_host
         self.vision_endpoint = vision_endpoint
         self.vision_token_provider = vision_token_provider
+        self.use_hugging_face = use_hugging_face
```

Comment:
Is this really needed? Do we need to set this field in the base `Approach` class?